### ISSUE TITLE
How to install modules?

### ISSUE THREAD

**Issue by dlmw** [azerothcore/acore-docker#41]
Hello, and first thank you for the amazing work in putting together those Docker images! My instance is running flawlessly.

There's this [mod-autobalance](https://www.azerothcore.org/catalogue#/details/76854669) module that I'd like to install and in the official docs they're talking about needing to recompile AzerothCore in order for it to work correctly. I'm not sure how that applies to Docker.

Could anybody help me figure it out? Thanks 💙

Edit: Worth noting I'm using Docker Swarm and not Docker Compose. I'm not sure how to translate `docker compose up -d --build` into Swarm.

**Comment by dezral** [comment]
I have not figured this out either. I would like to use the same mod and mod-individual-xp
https://github.com/azerothcore/mod-individual-xp.

but i think it's a volume we need to add to the composer file with the mods but not sure or had time to test it

**Comment by Helias** [comment]
indeed, you have to add a module that associate `your-module-directory` into `src/modules/`
hope to write a small documentation about it

**Comment by dezral** [comment]
> indeed, you have to add a module that associate `your-module-directory` into `src/modules/` hope to write a small documentation about it

im new to azerothcore but long time linux/docker user.. if you could point me in the right direction i could try write something down

**Comment by Helias** [comment]
Sorry, checking it better, probably you can't with the current docker image because the docker image includes the binary without any modules, they are just the results of the compilation of the azerothcore-wotlk source code, to add a module you should do a step back, so maybe use a docker image that replaces `ac-worldserver` with a docker image that includes the source code and it has a build step

**Comment by dezral** [comment]
> Sorry, checking it better, probably you can't with the current docker image because the docker image includes the binary without any modules, they are just the results of the compilation of the azerothcore-wotlk source code, to add a module you should do a step back, so maybe use a docker image that replaces `ac-worldserver` with a docker image that includes the source code and it has a build step

ahh okay thanks, i have been trying this below guide also, and it works until i get to the import the database and i get  this 
 ✘ Container ac-db-import         service "ac-db-import" didn't complete successfully: exit 1

I have not been able to find out why exactly , other than it have something to do with the connection to the database.. from log <Could not connect to MySQL database at ac-database: Access denied> and something that it try's to connect from wrong ip..  

think i just need to find out where to change the mysql user host from localhost to %. like here https://github.com/azerothcore/azerothcore-wotlk/tree/4db17e158c83e90c0fffddf8ab2dffdafd7545fd/data/sql/create

https://www.azerothcore.org/wiki/install-with-docker
git clone https://github.com/azerothcore/azerothcore-wotlk.git --depth 1
docker compose build
docker compose up -d

**Comment by grofit** [comment]
Did you ever get past this stage or did you give up? I have been asked to look into this and if you got it working yourself I would much rather use your work than have to tread the same ground again to get it working (funnily enough its the same mod as OP was wanting to install too)

**Comment by dezral** [comment]
no sorry gave up ;-)

i'm on the verge, to move my game server to a new proxmox server.. then i might be able to test things if you want

**Comment by grofit** [comment]
I ended up just using the official docker image via docker compose following this https://www.azerothcore.org/wiki/install-with-docker

My scenario was proxmox based too, but I have proxmox hosting 2 linux boxes, one of which has a USB passthrough for an external drive which is shared via Samba on the network. Unfortunately it was the other instance I wanted to run AC on so I cloned all the files onto the network drive (a NAS for all intents and purposes) and then used docker remote api (https://github.com/jarkt/docker-remote-api) hooked up to my 2nd docker box to relay the docker compose commmands over.

I had to do some fudgery in the `docker-compose.override.yml` to stop it looking for files on my NAS and override some basic config but I put the modules in the `modules` dir and the docker stuff built correctly and hosted fine, it was a bit more involved than I would like but works and it should withstand updates and more modules being added going forward.

**Comment by holliefae** [comment]
I've added this functionality in my Ansible scripts... it will clone the `azerothcore-wotlk` repo to build the images... then bring up the acore-docker compose. I think that's a similar workflow as to what happens here? https://github.com/assumptionsandg/ansible-collection-azerothcore
